goas/cl/ledger/src/partial_tx.rs

```diff
@@ -1,26 +1,65 @@
-use crate::{input::ProvedInput, output::ProvedOutput};
+use std::collections::BTreeMap;
+
+use proof_statements::death_constraint::DeathConstraintPublic;
+
+use crate::{death_constraint::DeathProof, input::ProvedInput, output::ProvedOutput};
+
+#[derive(Debug, Clone)]
+pub struct PartialTxInput {
+    pub input: ProvedInput,
+    pub death: DeathProof,
+}
+
+impl PartialTxInput {
+    fn verify(&self, ptx_root: cl::PtxRoot) -> bool {
+        let nf = self.input.input.input.nullifier;
+        self.input.input.input.death_cm == self.death.death_commitment() // ensure the death proof is actually for this input
```

Comment:
where do we check that the `death_cm` is consistent with the note commitment? in the input proof?

Reply:
yes, the input proof, through the `InputWitness::commit()` call: https://github.com/logos-co/nomos-pocs/blob/master/goas/cl/cl/src/input.rs#L61